pkgs/applications/video/gyroflow/default.nix

```diff
@@ -1,50 +1,67 @@
+  fetchFromGitHub,
+  makeDesktopItem,
+  clang,
+  copyDesktopItems,
+  patchelf,
+  pkg-config,
+  qt6,
+  alsa-lib,
+  bash,
+  ffmpeg,
+  mdk-sdk,
+  ocl-icd,
+  opencv,
 }:
-
+let
+  lens-profiles = fetchFromGitHub {
+    owner = "gyroflow";
+    repo = "lens_profiles";
+    tag = "v19";
+    hash = "sha256-8R2mMqKxzoa5Sfqxs8pcfwUfo1PQKSrnM+60Ri3wiXY=";
+  };
+in
 rustPlatform.buildRustPackage rec {
   pname = "gyroflow";
-  version = "1.5.4-2024-09-05";
+  version = "1.6.0";
 
   src = fetchFromGitHub {
     owner = "gyroflow";
     repo = "gyroflow";
-    rev = "52038dedad0bd14d6af68db36a09da0243ad5455";
-    hash = "sha256-EuhUF2b8YWv2eN2pcoHA0SlnyeQ8gJ5eHbXi6G4GIzk=";
+    tag = "v${version}";
+    hash = "sha256-Ib9GnHN23eTbd3nEwvZf3+CBSkUHycN77o3ura0Ze/0=";
   };
 
-  cargoLock = {
-    lockFile = ./Cargo.lock;
-    outputHashes = {
-      "ahrs-0.6.0" = "sha256-CxWyX8t+BjqIyNj1p1LdkCmNrtJkudmKgZPv0MVcghY=";
-      "akaze-0.7.0" = "sha256-KkGXKoVRZZ7HUTtWYBerrN36a7RqsHjYQb+bwG1JagY=";
-      "app_dirs2-2.5.5" = "sha256-nQ5Cs9r1k/3zjqXJ18Oilk8ErLKim7bGwCXDlQW4GRQ=";
-      "fc-blackbox-0.2.0" = "sha256-82DuI0KuHhDVhCMUsnDqk6Fk774VpvoJ1qYFLO+n1X4=";
-      "ffmpeg-next-7.0.4" = "sha256-F1N70tSxdC36uS9Bh7X2S9Wspd7bcCbGPmoMRs1cm8Y=";
-      "ffmpeg-sys-next-7.0.2" = "sha256-7C46WJseKIhqXW0cQGaF8Q/xQi7sX+e8fKVrhBMVwZE=";
-      "keep-awake-0.1.0" = "sha256-iZuntDkAhDZBojRgNEupAExtqOhiw4mf6XK0N6ff2Oc=";
-      "mp4parse-0.17.0" = "sha256-DktX6zmQsHBlo7uLgLXcXWxKq9uisnX0R16lftWRLZY=";
-      "naga-22.0.0" = "sha256-+ngomv0VyWKNDkSGVI/f5wtDyLs79qiXxtj3qNOsbFc=";
-      "nshare-0.9.0" = "sha256-PAV41mMLDmhkAz4+qyf+MZnYTAdMwjk83+f+RdaJji8=";
-      "qmetaobject-0.2.10" = "sha256-kEwFjDe1tYTLQ8XdjrPyYEYnjVFyYeaWUPCj5D8mg7A=";
-      "qml-video-rs-0.1.0" = "sha256-8RYB+numVy7u29EYtSSdf/+cTsUMVjrcw4u5mqB7KbE=";
-      "rs-sync-0.1.0" = "sha256-6xQb0CUlBDx7S7zsqNL9zfZZtkmw0cbUUXd6pOYIrXI=";
-      "spirv-std-0.9.0" = "sha256-uZn1p2pM5UYQKlY9u16aafPH7dfQcSG7PaFDd1sT4Qc=";
-      "system_shutdown-4.0.1" = "sha256-YypNnZzTxkmUgIxaP4jOpFBje/fEzI5L1g+3pJgMd0w=";
-      "telemetry-parser-0.3.0" = "sha256-U26cWC7pSw4NFiu43BZf+KlLy9NU61iRpFx3Btse1aY=";
-     };
-  };
+  useFetchCargoVendor = true;
 
-  lens-profiles = callPackage ./lens-profiles.nix { };
+  cargoHash = "sha256-bqBFAobXwPC4V0OYHbwmkk7shfiFt3YMGAf7F5ybLAQ=";
 
   nativeBuildInputs = [
-    clang copyDesktopItems patchelf pkg-config rustPlatform.bindgenHook wrapQtAppsHook
+    clang
+    copyDesktopItems
+    patchelf
+    pkg-config
+    rustPlatform.bindgenHook
+    qt6.wrapQtAppsHook
   ];
 
-  buildInputs = [ alsa-lib bash ffmpeg mdk-sdk ocl-icd opencv qtbase qtdeclarative qtsvg ];
+  buildInputs = [
+    alsa-lib
+    bash
+    ffmpeg
+    mdk-sdk
+    ocl-icd
+    opencv
+    qt6.qtbase
+    qt6.qtdeclarative
+    qt6.qtsvg
+  ];
 
-  patches = [ ./no-static-zlib.patch ];
+  postPatch = ''
+    sed -i '/println!("cargo:rustc-link-lib=static:+whole-archive=z")/d' build.rs
```

Comment:
Nixpkgs discourage using `sed`. With `sed` it is not obvious which changes are applied and which are obsolete and do nothing, and when updating the package it is hard to notice that the change needs updating too. This is why Nixpkgs use `substituteInPlace` (https://nixos.org/manual/nixpkgs/stable/#fun-substitute) or patch files. I would have been OK with `substituteInPlace --replace-fail` here, but IMHO patch is clearer (because it explicitly lists all the occurrences of the change) and easier to maintain (I just edit the file under git control and export the diff)